checking/integration/support/e2eservice.go

```diff
@@ -0,0 +1,181 @@
+package support
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/optimizely/go-sdk/checking/integration/optimizely/datamodels"
+	"github.com/optimizely/go-sdk/checking/integration/optimizely/eventdispatcher"
+	"github.com/optimizely/go-sdk/checking/integration/optimizely/listener"
+	"github.com/optimizely/go-sdk/optimizely/client"
+	"github.com/optimizely/go-sdk/optimizely/config"
+	"github.com/optimizely/go-sdk/optimizely/decision"
+	"github.com/optimizely/go-sdk/optimizely/entities"
+	"github.com/optimizely/go-sdk/optimizely/event"
+	"github.com/optimizely/go-sdk/optimizely/utils"
+	"gopkg.in/yaml.v3"
+)
+
+func setupOptimizelyClient(requestParams *datamodels.RequestParams) {
```

Comment:
This is a bit convoluted. It should be reading the API options passed in and returning a client wrapper instance instead of writing the client into the param object that was passed in